build/bin/network-validator.go

```diff
@@ -100,17 +105,55 @@ func TestEndpoints(config reachabilityConfig) {
 	os.Exit(0)
 }
 
-func ValidateReachability(host string, port int, tlsDisabled bool) error {
+func embedProxyCertificate(cacertFile string) (*x509.CertPool, error) {
+	// Get the SystemCertPool, continue with an empty pool on error
+	rootCAs, _ := x509.SystemCertPool()
+	if rootCAs == nil {
+		rootCAs = x509.NewCertPool()
+	}
+
+	if cacertFile == "" {
+		return rootCAs, nil
+	}
+
+	// Read in the cert file
+	certs, err := ioutil.ReadFile(cacertFile)
+	if err != nil {
+		return nil, err
+	}
+
+	// Append our cert to the system pool
+	if ok := rootCAs.AppendCertsFromPEM(certs); !ok {
```

Comment:
note to myself: make sure this is not continuously adding the same cert to the pool for each validation

Reply:
rootCA is just a copy/value of x509.SystemCertPool() and being passed with customca individually for each endpoint